src/main/java/teammates/ui/controller/StudentProfilePictureUploadAction.java

```diff
@@ -42,12 +44,15 @@ protected ActionResult execute() throws EntityDoesNotExistException {
             if (!isError) {
                 blobKey = blobInfo.getBlobKey();
                 pictureKey = renameFileToGoogleId(blobInfo);
-                logic.updateStudentProfilePicture(account.googleId, pictureKey);
+                logic.updateOrCreateStudentProfile(
```

Comment:
However, profile picture will be updated in this case and other attributes will be leaven untouched.